apps/system/js/app_usage_metrics.js

```diff
@@ -850,6 +895,24 @@
         }
       }
     }
+
+    // loop through all the search providers that we have data for
+    // and merge the new searches into the old searches.
+    for (var provider in newbatch.data.searches) {
+      var newsearch = newbatch.data.searches[provider];
+      var oldsearch = this.data.searches[provider];
+
+      for (var daykey in newsearch) {
+        var daySearch = oldsearch[daykey];
```

Comment:
Verify that `oldsearch` actually exists here.. it's not guaranteed that a new batch of data will have the same set of search providers as an old set is it?  Here's what I'm thinking:

I submit a search using yahoo, then the batch of AppUsageMetrics is saved when I turn off my phone. I turn it on again, and search with google this time, but google doesn't exist in the first batch.. so `oldsearch` here would actually be undefined. does that sound plausible?